packages/cli/src/spec/mock/MockPromptForStrings.ts

```diff
@@ -0,0 +1,107 @@
+
+  /**
+   * Sets the output to be the given message.
+   * @param message the message to be prompt. Not relevant for tests.
+   * @param {string | null} defaultValue a default value or null.
+   * @param {ValidateFunction<T>} validateFunction a function to validate the input.
+   * @returns {Promise<T>} a {@link Promise} that resolves to the validated loaded message,
+   * converted to `T` by the `validateFunction`.
+   */
+  async promptInput<T>(_message: string,
+      _defaultValue: string | null,
+      validateFunction: ValidateFunction<T>): Promise<T> {
+    const nextResponse = this.getNextMessage();
+    return (await validateFunction(nextResponse)).unwrap();
+  }
+
+  /**
+   * Sets the output to be the given message.
+   * @param message the message to be prompt. Not relevant for tests.
+   * @param {string[]} choices a list of choices. Not relevant for testing.
+   * @param {string | null} defaultValue a default value or null.
+   * @param {ValidateFunction<T>} validateFunction a function to validate the input.
+   * @returns {Promise<T>} a {@link Promise} that resolves to the validated loaded message,
+   * converted to `T` by the `validateFunction`.
+   */
+  async promptChoice<T>(_message: string,
+      _choices: string[],
+      _defaultValue: string | null,
+      validateFunction: ValidateFunction<T>): Promise<T> {
+    const nextResponse = this.getNextMessage();
+    return (await validateFunction(nextResponse)).unwrap();
+  }
+
+  /**
+   * Sets the output to be the given message.
+   * @param message the message to be prompt. Not relevant for tests.
+   * @param defaultValue the value to be returned
+   * @returns {Promise<boolean>} a {@link Promise} that resolves to a {@link boolean} value. The
+   * value will the `true` if the user answers `Yes` and `false` for `No`.
+   */
+  async promptConfirm(_message: string, defaultValue: boolean): Promise<boolean> {
+    return defaultValue;
+  }
+
+  /**
+   * Sets the output to be the given message.
+   * @param message the message to be prompt. Not relevant for tests.
+   * @param {ValidateFunction<T>} validateFunction a function to validate the input.
+   * @returns {Promise<string>} a {@link Promise} that resolves to the user input validated by
+   * `validateFunction`.
+   */
+  async promptPassword(_message: string, validateFunction: ValidateFunction<string>,
+  ): Promise<string> {
+    const nextResponse = this.getNextMessage();
+    return (await validateFunction(nextResponse)).unwrap();
+  }
+
+  /**
+   * Sets the output to be the given message.
+   * @param {ValidateFunction<T>} validateFunction a function to validate the input.
+   * @returns {string} which is the next message to be prompted`.
+   */
+  private getNextMessage(): string {
+    if (this.responses.length < 0) {
+      throw new Error('No answer was given. Please use addMessage(NextResponse) before' +
+      ' using this function');
+    }
+    const nextResponse = this.responses[this.responses.length];
```

Comment:
You're adding responses with `this.responses.push(...)` which adds them to the end of the queue, then you're getting the next response, which is the last one, then you're calling pop, which removes the last element of the queue. I think the order is going to get messed up, for example:

1. Call `addMessage("hello")`, `responses = ["hello"]`.
2. Call `addMessage("world")`, `responses = ["hello", "world"]`.
3. Call `getNextMessage()`, it returns `"world"`, `responses = ["hello"]`.
4. Call `getNextMessage()`, it returns `"hello"`, `responses = []`.

So you're return the responses in reverse order.

Here are the [methods you have on an array][1] to help you find what operations to perform.

[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array